src/fns.rs

```diff
@@ -615,6 +615,187 @@ pub(crate) fn elt<'ob>(sequence: Object<'ob>, n: usize, cx: &'ob Context) -> Res
     }
 }
 
+#[defun]
+pub(crate) fn string_equal<'ob>(s1: Object<'ob>, s2: Object<'ob>) -> Result<Option<bool>> {
+    let s1 = match s1.untag() {
+        ObjectType::String(x) => x.as_bytes(),
+        ObjectType::ByteString(x) => x.inner(),
+        ObjectType::Symbol(x) => x.get().as_bytes(),
+        _ => bail!("Wrong type argument: stringp, {s1}"),
+    };
+    let s2 = match s2.untag() {
+        ObjectType::String(x) => x.as_bytes(),
+        ObjectType::ByteString(x) => x.inner(),
+        ObjectType::Symbol(x) => (x.get()).as_bytes(),
+        _ => bail!("Wrong type argument: stringp, {s2}"),
+    };
+
+    if s1.len() != s2.len() {
+        return Ok(None);
+    }
+    if s1 == s2 {
+        Ok(Some(true))
+    } else {
+        Ok(None)
+    }
+}
+
+#[defun]
+pub(crate) fn compare_strings<'ob>(
+    string1: Object<'ob>,
```

Comment:
We should be able to use the more specific type of `string1: &str` here to make the code cleaner